### PR TITLE
see #1043 Fix marble diagrams

### LINKED ISSUE #1043: 3.1.x marble diagram issues
This issue tracks marble diagram related problems. Which can be:

 - out of date marble diagrams (the operator has changed but not the diagram)
 - wrong link used
 - operator has been renamed / aliased in 3.1.0 but the diagram hasn't been updated
 - new operator is missing a diagram
 - diagram PNG was edited but the OmniGraffle hasn't

Sub-issues can be attached to this tracking issue.

# Marble Diagram Issues

 - [ ] `then`: old operator named used on `Mono.thenEmpty(Publisher<Void>)` and `Mono.thenMany(Publisher<V>)` (#1008)
 - [ ] `windowUntil`: `windowUntil.png` should basically be the same as `windowUntilCutBefore.png` minus the operator block.
 - [ ] Missing diagrams for `concatWithValues`, `thenReturn` and `mergeOrdered`/`mergeOrderedWith`
 - [ ] `Mono.when*` and `Mono.zip*` (#1167)
 - [ ] `Flux.onErrorResume` shows `onErrorResumeWith` in its marble

### PR DESCRIPTION
**Pull request by madgnome**
#### Fix windowUntil marble

It should basically be the same as windowUntilCutAfter.png minus the operator block.

The comment on the issue https://github.com/reactor/reactor-core/issues/1043 says it should be like `windowUntilCutBefore.png` but looking at the code `windowUntil` is an alias for `windowUntil(Predicate, cutBefore=false)`

#### Add missing diagrams

- `Mono.thenReturn`
- `Flux.concatWithValues`
- `Flux.mergeOrdered`/`Flux.mergeOrderedWith`

#### Fix Mono.when* and Mono.zip* diagrams

### CODE REVIEW

**Comment by simonbasle** [comment]
@madgnome do you want to try and improve more of these diagrams in this PR? 😍

**Comment by madgnome** [comment]
I may as well since I already implied I will fix the whole issue in my commit message 😄

File: README.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ Flux.fromIterable(getSomeLongList())
     .doOnNext(serviceA::someObserver)
     .map(d -> d * 2)
     .take(3)
-    .onErrorResumeWith(errorHandler::fallback)
+    .onErrorResume(errorHandler::fallback)
```
Comment (madgnome) [inline README.md]: Fixed README using the old operator name.

**Comment by madgnome** [comment]
Fix all tasks currently listed on #1043

**Comment by madgnome** [comment]
I've updated the urls to point to master and raised an issue so we remember to clean them up later https://github.com/reactor/reactor-core/issues/1227

File: reactor-core/src/main/java/reactor/core/publisher/Flux.java [hunk 2]
```diff
@@ -5844,7 +5858,7 @@ public int getPrefetch() {
 	 * Subscribe to a fallback publisher when an error matching a given predicate
 	 * occurs.
 	 * <p>
-	 * <img class="marble" src="https://raw.githubusercontent.com/reactor/reactor-core/v3.1.3.RELEASE/src/docs/marble/onerrorresumewith.png"
+	 * <img class="marble" src="https://raw.githubusercontent.com/reactor/reactor-core/v3.1.3.RELEASE/src/docs/marble/onerrorresume.png"
```
Comment (simonbasle) [inline reactor-core/src/main/java/reactor/core/publisher/Flux.java]: missed this `v3.1.3.RELEASE` to `master`

File: reactor-core/src/main/java/reactor/core/publisher/Flux.java [hunk 3]
```diff
@@ -5822,7 +5836,7 @@ public int getPrefetch() {
 	 * Subscribe to a fallback publisher when an error matching the given type
 	 * occurs, using a function to choose the fallback depending on the error.
 	 * <p>
-	 * <img class="marble" src="https://raw.githubusercontent.com/reactor/reactor-core/v3.1.3.RELEASE/src/docs/marble/onerrorresumewith.png"
+	 * <img class="marble" src="https://raw.githubusercontent.com/reactor/reactor-core/v3.1.3.RELEASE/src/docs/marble/onerrorresume.png"
```
Comment (simonbasle) [inline reactor-core/src/main/java/reactor/core/publisher/Flux.java]: missed this `v3.1.3.RELEASE` to `master`

File: reactor-core/src/main/java/reactor/core/publisher/Flux.java [hunk 4]
```diff
@@ -5808,7 +5822,7 @@ public int getPrefetch() {
 	 * choose the fallback depending on the error.
 	 *
 	 * <p>
-	 * <img class="marble" src="https://raw.githubusercontent.com/reactor/reactor-core/v3.1.3.RELEASE/src/docs/marble/onerrorresumewith.png" alt="">
+	 * <img class="marble" src="https://raw.githubusercontent.com/reactor/reactor-core/v3.1.3.RELEASE/src/docs/marble/onerrorresume.png" alt="">
```
Comment (simonbasle) [inline reactor-core/src/main/java/reactor/core/publisher/Flux.java]: missed this `v3.1.3.RELEASE` to `master`

File: reactor-core/src/main/java/reactor/core/publisher/Flux.java [hunk 5]
```diff
@@ -5432,6 +5443,9 @@ public int getPrefetch() {
 	 * an infinite source that doesn't already publish on a dedicated Scheduler, you must isolate that source
 	 * in its own Scheduler, as merge would otherwise attempt to drain it before subscribing to
 	 * another source.
+	 * <p>
```
Comment (simonbasle) [inline reactor-core/src/main/java/reactor/core/publisher/Flux.java]: missed this `v3.1.3.RELEASE` to `master`

File: reactor-core/src/main/java/reactor/core/publisher/Mono.java [hunk 6]
```diff
@@ -1013,7 +1013,7 @@
 	 * to be cancelled and the resulting Mono to immediately error or complete, respectively.
 	 *
 	 * <p>
-	 * <img class="marble" src="https://raw.githubusercontent.com/reactor/reactor-core/v3.1.3.RELEASE/src/docs/marble/whent.png" alt="">
+	 * <img class="marble" src="https://raw.githubusercontent.com/reactor/reactor-core/v3.1.3.RELEASE/src/docs/marble/zip1.png" alt="">
```
Comment (simonbasle) [inline reactor-core/src/main/java/reactor/core/publisher/Mono.java]: missed this `v3.1.3.RELEASE` to `master`

File: reactor-core/src/main/java/reactor/core/publisher/Mono.java [hunk 7]
```diff
@@ -1033,7 +1033,7 @@
 	 * An error or <strong>empty</strong> completion of any source will cause other sources
 	 * to be cancelled and the resulting Mono to immediately error or complete, respectively.
 	 * <p>
-	 * <img class="marble" src="https://raw.githubusercontent.com/reactor/reactor-core/v3.1.3.RELEASE/src/docs/marble/whent.png" alt="">
+	 * <img class="marble" src="https://raw.githubusercontent.com/reactor/reactor-core/v3.1.3.RELEASE/src/docs/marble/zip1.png" alt="">
```
Comment (simonbasle) [inline reactor-core/src/main/java/reactor/core/publisher/Mono.java]: missed this `v3.1.3.RELEASE` to `master`

File: reactor-core/src/main/java/reactor/core/publisher/Mono.java (hunk 6 above)
Comment (madgnome) [reply reactor-core/src/main/java/reactor/core/publisher/Mono.java]: Actually this one already exists in `v3.1.3.RELEASE`

Comment (simonbasle) [reply reactor-core/src/main/java/reactor/core/publisher/Mono.java]: :+1:

**Pull request merged**